kubeconfig-creator/README.md

```diff
@@ -0,0 +1,155 @@
+are intended as its replacement. This is part of our plan to [offer replacement
+`tasks` for Pipeline Resources](https://github.com/tektoncd/catalog/issues/95)
+as well as
+[document those replacements](https://github.com/tektoncd/pipeline/issues/1369).
+
+
+This `task` creates a [kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/)
+file that can be used to configure access to the different clusters.
+A common use case for this task is to deploy your `application/function` on different clusters.
+
+The task will use the [kubeconfigwriter](https://github.com/Divyansh42/kubeconfigwriter) image and the provided parameters to create a `kubeconfig` file that can be used by other tasks
+in the pipeline task to access the target cluster. The kubeconfig will be placed at 
+`/workspace/kubeconfigFile/kubeconfig`.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/kubeconfig-creator/kubeconfig-creator.yaml
+
+```
+## Workspace
+
+* **kubeconfigFile**: A workspace that stores the generated kubeconfig file, such that it can be used in the other tasks to access the cluster.
+
+## Parameters
+
+* **Name**: Name of the `cluster`.
+* **Type**: Type of the resource (_default:_
+  `cluster`)
+* **URL**: Address of the target cluster (_e.g.:_ 
+  `https://hostname:port`)
+* **Username**: Username for basic authentication to the cluster
+(_default:_ `""`)
+* **Password**: Password for basic authentication to the cluster
+(_default:_ `""`)
+* **Cadata**: Contains PEM-encoded certificate authority certificates
+(_default:_ `""`)
+* **Namespace**: Default namespace to use on unspecified requests
+(_default:_ `""`)
+* **Token**: Bearer token for authentication to the cluster
+(_default:_ `""`)
+* **Insecure**:  If true, skips the validity check for the server's certificate. 
+This will make your HTTPS connections insecure
+(_default:_ `false  `)
+
+
+## Usage
+
+This task uses a `shared workspace` with [`PVC`](https://kubernetes.io/docs/concepts/storage/persistent-volumes) to store the
+ `kubeconfig` in the `kubeconfigFile` directory. 
+Kubeconfig file is stored at `/workspace/kubeconfigFile/kubeconfig`.
+
+Task can be used with the other task in the pipeline to authenticate the cluster.
+In this example, pipeline has a task `kubeconfig-creator` that generates a 
+`kubeconfig file` for the cluster and the `test-task` uses that kubeconfig file and verifiy whether the
+application has an access to the cluster or not by using some `kubectl/oc` commands.
+
+```
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: kubeconfig-test-pipeline
+spec:
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - name: kubeconfig-creator
+      taskRef:
+        name: kubeconfig-creator
+      workspaces:
+        - name: kubeconfigFile
```

Comment:
it might be a little clearer if we just named this workspace `output`?  I got a bit confused to see a directory called `kubeconfigFile` mentioned everywhere in this doc.

Reply:
Yes, thanks for the suggestion.